xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/test/java/org/xwiki/rendering/macro/descriptor/DefaultMacroDescriptorTest.java

```diff
@@ -19,229 +19,212 @@
-
-        public String getUpperParam()
-        {
-            return this.upperParam;
-        }
-
-        @PropertyDescription("param1 description")
-        public void setParam1(String param1)
-        {
-            this.param1 = param1;
-        }
-
-        public String getParam1()
-        {
-            return this.param1;
-        }
-
-        @PropertyMandatory
-        @PropertyDescription("param2 description")
-        public void setParam2(int param1)
-        {
-            this.param2 = param1;
-        }
-
-        public int getParam2()
-        {
-            return this.param2;
-        }
-
-        public void setParam3(boolean param1)
-        {
-            this.param3 = param1;
-        }
-
-        @PropertyMandatory
-        @PropertyDescription("param3 description")
-        public boolean getParam3()
-        {
-            return this.param3;
-        }
-
-        @PropertyHidden
-        public void setHiddenParameter(String hiddenParameter)
-        {
-            this.hiddenParameter = hiddenParameter;
-        }
-
-        public String getHiddenParameter()
-        {
-            return this.hiddenParameter;
-        }
-
-        @Deprecated
-        public String getDeprecatedParameter()
-        {
-            return deprecatedParameter;
-        }
-
-        @Deprecated
-        public void setDeprecatedParameter(String deprecatedParameter)
-        {
-            this.deprecatedParameter = deprecatedParameter;
-        }
-
-        @PropertyAdvanced
-        public String getAdvancedParameter()
-        {
-            return advancedParameter;
-        }
-
-        public void setAdvancedParameter(String advancedParameter)
-        {
-            this.advancedParameter = advancedParameter;
-        }
-    }
+    @InjectMockComponents
+    private DefaultBeanManager beanManager;
 
-    private DefaultMacroDescriptor macroDescriptor;
-
-    @Before
-    @Override
-    public void setUp() throws Exception
+    @Test
+    public void parameterDescriptorTests()
     {
-        super.setUp();
-        BeanManager propertiesManager = getComponentManager().getInstance(BeanManager.class);
-        this.macroDescriptor =
-            new DefaultMacroDescriptor(new MacroId("Id"), "Name", "Description", new DefaultContentDescriptor(),
-                propertiesManager.getBeanDescriptor(ParametersTests.class));
+        DefaultMacroDescriptor macroDescriptor =
+                new DefaultMacroDescriptor(new MacroId("Id"), "Name", "Description", new DefaultContentDescriptor(),
+                        this.beanManager.getBeanDescriptor(ParametersTests.class));
+        Map<String, ParameterDescriptor> map = macroDescriptor.getParameterDescriptorMap();
+
+        assertNull(map.get("hiddenparameter"));
+
+        getId(map);
+        getDescription(map);
+        getParameterType(map);
+        getDefaultValue(map);
+        isMandatory(map);
+        isDeprecated(map);
+        isAdvanced(map);
+        getGroup(map);
```

Comment:
This isn't too good :) Best practice is to have a test method per tested feature. For unit tests we must have the max test methods to isolate problems.